library/core/class.oauth2.php

```diff
@@ -220,13 +222,15 @@ public function setRequestAccessTokenParams($params) {
 
 
     /**
-     * Set additional params to be added to the get string in the getProfile request.
+     * Set additional params to be to be merged with the default parameters
+     * in the getProfile request.
+     *
+     * @param array $params Params to add to the get profile request.
      *
-     * @param string $params.
      * @return $this Return this object for chaining purposes.
      */
-    public function setGetProfileParams($params) {
-        $this->getProfileParams = $params;
+    public function setRequestProfileParams($params = []) {
```

Comment:
The function should not be called without a parameter since it is a setter.
The DocBlock says that the parameter needs to be an array. 

That should be enough.